core/drivers/clk/clk.c

```diff
@@ -17,22 +17,28 @@ static unsigned int clk_lock = SPINLOCK_UNLOCK;
 struct clk *clk_alloc(const char *name, const struct clk_ops *ops,
 		      struct clk **parent_clks, size_t parent_count)
 {
-	struct clk *clk = NULL;
+	struct clk_elt *clk_elt = NULL;
 	size_t parent = 0;
 
-	clk = calloc(1, sizeof(*clk) + parent_count * sizeof(clk));
-	if (!clk)
+	if (ops->id != CLK_OPS_ELT) {
```

Comment:
I suppose an assert to help while debugging could be justified, but this seems a bit much to me.

Reply:
the function registers a clk_elt instance. If not of the right type, clk_alloc() should fail. Since it provides or errror code, returning NULL would tends to report heap exhaustion. I'll switch to an assertion.